src/main/java/org/candlepin/subscriptions/resource/CapacityResource.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.resource;
+
+import org.candlepin.subscriptions.db.model.TallyGranularity;
+import org.candlepin.subscriptions.utilization.api.model.CapacityReport;
+import org.candlepin.subscriptions.utilization.api.model.CapacityReportMeta;
+import org.candlepin.subscriptions.utilization.api.model.CapacitySnapshot;
+import org.candlepin.subscriptions.utilization.api.model.TallyReportLinks;
+import org.candlepin.subscriptions.utilization.api.resources.CapacityApi;
+
+import org.springframework.stereotype.Component;
+
+import java.time.OffsetDateTime;
+import java.time.Period;
+import java.time.temporal.TemporalAmount;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Capacity API implementation.
+ */
+@Component
+public class CapacityResource implements CapacityApi {
+
+    @Override
+    public CapacityReport getCapacityReport(String productId, @NotNull String granularity,
+        @NotNull OffsetDateTime beginning, @NotNull OffsetDateTime ending, Integer offset, Integer limit) {
+
+        return new CapacityReport()
+            .data(createMockData(granularity, beginning, ending))
+            .links(createMockLinks())
+            .meta(createMockMeta(productId, granularity, beginning, ending));
+    }
+
+    private CapacityReportMeta createMockMeta(String productId, @NotNull String granularity,
+        @NotNull OffsetDateTime beginning, @NotNull OffsetDateTime ending) {
+
+        return new CapacityReportMeta()
+            .count(createMockData(granularity, beginning, ending).size())
```

Comment:
Any reason to create the data again just to get the size? Can we just pass the snapshot count, productid and granularity?

Reply:
@mstead, addressed!